backend/data_preparation/crawler/fire_crawler.py

```diff
@@ -0,0 +1,318 @@
+        """
+        if not ".zip" == single_file_name[-4:]:
+            # if it is a zip don't download
+            folder_name = single_file_name.split(".")[0]
+            if folder_name not in used_folder_names:
+                # if the folder name is not in used_folder_names, which means it is a new folder
+                # then create the folder and add it into the used_folder_names set, to avoid duplicate folders
+                used_folder_names.add(folder_name)
+                os.makedirs(os.path.join(FIRE_DATA_DIR, folder_name))
+            out_path = os.path.join(FIRE_DATA_DIR, folder_name)
+            FireCrawler._download_single_file(url_of_fire_event, single_file_name, out_path)
+
+    @staticmethod
+    @retry(retry_on_exception=requests.exceptions.RequestException, stop_max_attempt_number=3)
+    def _extract_file_names_in_fire_event(url_of_fire_event: str) -> List[str]:
+        """
+        Extracts all file names from a fire_event page. Return a list of all file names.
+        :param url_of_fire_event: the url of the fire_event
+                e.g. https://rmgsc.cr.usgs.gov/outgoing/GeoMAC/2017_fire_data/California/Antelope/
+        :return: e.g. ["ca_antelope_20170706_1930_dd83.CPG", "ca_antelope_20170706_1930_dd83.dbf"]
+        """
+        fire_page = FireCrawler._get_url(url_of_fire_event, url_of_fire_event, "crawl()")
+        return FireCrawler.RE_EXTRACT_FILES_IN_FIRE_EVENTS.findall(fire_page)
+
+    def crawl(self, url_of_fire_event: str):
+        """
+        Takes the url string of one fire and start to crawl all files.
+        :param url_of_fire_event: str
+                e.g. "https://rmgsc.cr.usgs.gov/outgoing/GeoMAC/2016_fire_data/California/Happy_Camp"
+        """
+        used_folder_names = set()
+        # grab all fire names in the page
+        # for loop for runtime error
+        # download useful files
+        for file in FireCrawler._extract_file_names_in_fire_event(url_of_fire_event):
+            self._download_fire_record(file, used_folder_names, url_of_fire_event)
+        return
+
+    @staticmethod
+    def cleanup():
+        """
+        Cleans up the temp data folder
+        """
+        logger.info("Cleaning up the temp data folder...")
+        folder_names = glob.glob(os.path.join(FIRE_DATA_DIR, "[a-z]*"))
+        for folder_name in folder_names:
+            shutil.rmtree(folder_name)
+        logger.info("Finished cleaning up the temp data folder.")
+
+
+if __name__ == '__main__':
+    # module tester
+    logger.setLevel(logging.INFO)
+    logger.addHandler(logging.StreamHandler())
+    test_crawler = FireCrawler(["California"])
+    print(list(map(lambda fire: str(fire), test_crawler.extract_all_fires(2015))))
+    used = set()
+    test_crawler.crawl("https://rmgsc.cr.usgs.gov/outgoing/GeoMAC/current_year_fire_data/California/Trestle/")
+    test_crawler.cleanup()
```

Comment:
maybe run a `cleanup()` in `__init__()` and `crawl()`? out-scope might not know when should call `cleanup()`